app/src/main/java/cm/aptoide/pt/app/view/AppViewPresenter.java

```diff
@@ -138,6 +138,96 @@ public AppViewPresenter(AppViewView view, AccountNavigator accountNavigator,
     handleMoPubConsentDialog();
   }
 
+  private Observable<AppViewModel> loadAppView() {
+    return appViewManager.getAppViewModel()
+        .observeOn(viewScheduler)
+        .doOnSuccess(this::showAppView)
+        .doOnSuccess(this::sendAppViewLoadAnalytics)
+        .toObservable()
+        .flatMap(this::handleAppViewOpenOptions)
+        .filter(model -> !model.getAppModel()
+            .hasError())
+        .flatMap(this::loadOtherAppViewComponents);
+  }
+
+  private void showAppView(AppViewModel appViewModel) {
+    if (appViewModel.getAppModel()
+        .hasError()) {
+      view.handleError(appViewModel.getAppModel()
+          .getError());
+    } else {
+      view.showAppView(appViewModel.getAppModel());
+      view.showDownloadAppModel(appViewModel.getDownloadModel(),
+          appViewModel.getAppCoinsViewModel());
+      if (appViewModel.getAppCoinsViewModel()
+          .hasAdvertising() || appViewModel.getAppCoinsViewModel()
+          .hasBilling()) {
+        view.setupAppcAppView();
+      }
+      view.recoverScrollViewState();
+      view.readyToDownload();
+    }
+  }
+
+  private void sendAppViewLoadAnalytics(AppViewModel appViewModel) {
+    AppModel appModel = appViewModel.getAppModel();
+    if (!appModel.getEditorsChoice()
+        .isEmpty()) {
+      appViewManager.sendEditorsChoiceClickEvent(appModel.getPackageName(),
+          appModel.getEditorsChoice());
+    }
+    appViewManager.sendAppViewOpenedFromEvent(appModel.getPackageName(), appModel.getDeveloper()
+        .getName(), appModel.getMalware()
+        .getRank()
+        .name(), appModel.hasBilling(), appModel.hasAdvertising());
+
+    if (appViewModel.getDownloadModel()
+        .getAction()
+        .equals(DownloadModel.Action.MIGRATE) && !appViewManager.isMigrationImpressionSent()) {
+      appViewManager.setMigrationImpressionSent();
+      appViewAnalytics.sendAppcMigrationAppOpen();
+    }
+  }
+
+  private Observable<AppViewModel> handleAppViewOpenOptions(AppViewModel appViewModel) {
+    AppModel appModel = appViewModel.getAppModel();
+    DownloadModel.Action action = appViewModel.getDownloadModel()
+        .getAction();
+    return accountManager.accountStatus()
+        .first()
+        .observeOn(viewScheduler)
+        .flatMap(__ -> handleOpenAppViewDialogInput(appViewModel.getAppModel()))
+        .filter(shouldDownload -> shouldDownload)
```

Comment:
In a normal click to open app view with OPEN_ONLY OpenType, shouldDownload will be false and nothing else in the chain will happen, which it shouldn't for the downloadApp and so on but when you go back to the loadAppView() the loadOtherAppViewComponents(appModel)  method isn't called. This should be the reason why the similar apps, reviews aren't being shown. The banner, legacy ads and donations should also not show up because of this, but didn't test.

Reply:
This should now be fixed 👍 I've added tests to verify that these methods are being called so that this won't happen again.